README.md

```diff
@@ -230,6 +296,26 @@ and `autoValue` that performs the following steps:
 1. If `value` is `undefined`, return `0n`.
 1. Otherwise, return `value + 1n`.
 
+## ADT Enum Member Factories
+
+An ADT enum member can produce an enum member factory, which is either a `function` (in the case of `enum E { X(y) }`),
+or an `object` with a `@@propertyConstruct` property (in the case of `enum E { X{ y } }`).
+
+## Object Literal-like Construction Syntax
+
+This is intended to dovetail with the [Extractors proposal](https://github.com/tc39/proposal-extractors), and uses the
+@@propertyConstruct built-in symbol. The following syntax:
+
+```js
+const msg = Message.Move{ x, y };
+```
+
+is transposed the following representation:
+
+```js
+const msg = Message.Move[Symbol.propertyConstruct](#{ x, y });
```

Comment:
do we need R&T here?

Reply:
I think so, yes. I'm of the opinion that ADT enum should have the same requirements as records and tuples as to the kinds of values they can hold. One reason is data portability between realms, along with the potential for a future `shared enum` that could tie into `shared struct` and multi-threaded JS. I'd also like to have `Option.Some(1) === Option.Some(1)` hold.